tdxhydrorapid/weights.py

```
@@ -202,33 +222,107 @@ def apply_weight_table_simplifications(save_dir: str,
 
+    # group by matching values in columns except for area_sqm, and sum the areas in grouped rows
     wt = wt.groupby(wt.columns.drop('area_sqm').tolist()).sum().reset_index()
     wt = wt.sort_values([id_field, 'area_sqm'], ascending=[True, False])
     wt.to_csv(weight_table_out_path, index=False)
     return
+
+def get_weight_table_name_from_grid(grid: Union[str, gpd.GeoDataFrame], warn=False) -> str:
+    if isinstance(grid, str):
+        gpq = gpd.read_parquet(grid)
+    else:
+        gpq = grid
```

Ruff (0.8.0)

Use ternary operator `gpq = gpd.read_parquet(grid) if isinstance(grid, str) else grid` instead of `if`-`else`-block

Replace `if`-`else`-block with `gpq = gpd.read_parquet(grid) if isinstance(grid, str) else grid`

(SIM108)